src/db.rs

```diff
@@ -146,47 +207,78 @@ pub fn get_secret_policy(sec: &str) -> Option<policy::Policy> {
     })
 }
 
-pub fn insert_polid(sec: &str, polid: u32) -> Result<u32> {
-    let mut dbconn = get_dbconn().unwrap();
+pub fn insert_keyset(ksetid: &str, kskeys: &[String], polid: u32) -> Result<()> {
+    let mut dbconn = get_dbconn()?;
 
-    let mqstr = "INSERT INTO secpol (secret, polids ) VALUES(?, ?)".to_string();
+    let mysqlstr = "INSERT INTO keysets (keysetid, kskeys, polid)
+                    VALUES(?, ?, ?)"
+        .to_string();
 
     let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
+    let kskeys_str = serde_json::to_string(kskeys)?;
 
-    trnsx.exec_drop(mqstr, (&sec, &polid))?;
-    trnsx.commit()?;
-    Ok(polid)
+    trnsx.exec_drop(mysqlstr, (ksetid, &kskeys_str, polid))?;
+
+    Ok(trnsx.commit()?)
 }
 
-pub fn delete_polid(sec: &str) -> Result<String> {
-    let mut dbconn = get_dbconn().unwrap();
+pub fn delete_keyset(ksetid: &str) -> Result<()> {
+    let mut dbconn = get_dbconn()?;
 
-    let mqstr = "DELETE from secpol WHERE secret = ?";
+    let mysqlstr = "DELETE from keysets WHERE keysetid = ?".to_string();
 
     let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
 
-    trnsx.exec_drop(mqstr, (&sec,))?;
-    trnsx.commit()?;
-    Ok(sec.to_string())
+    trnsx.exec_drop(mysqlstr, (ksetid,))?;
+
+    Ok(trnsx.commit()?)
```

Comment:
I find this clearer since I assume `commit()` doesn't return any value inside its Result:

```rust
trnsx.exec_drop(mysqlstr, (ksetid,))?;
trnsx.commit()?;
Ok(())
```

Reply:
Fixed.

Reply:
hmm, I still see the original code. Did you `git add` all the changes?